src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java

```diff
@@ -238,9 +236,21 @@ public HashMap<Integer,HashMap<Integer, String>> getGeneticAlterationMapForEntit
                 HashMap<Integer, String> mapSampleValue = new HashMap<Integer, String>();
                 int geneticEntityId = rs.getInt("GENETIC_ENTITY_ID");
                 String values = rs.getString("VALUES");
-                //hm.debug..
-                String valueParts[] = values.split(DELIM);
-                for (int i=0; i<valueParts.length; i++) {
+                String[] valueParts = values.split(DELIM, -1);
+                int valuesLength = valueParts.length;
+                boolean hasMeaninglessTrailingDelimiter = valuesLength - orderedSampleList.size() == 1 && valueParts[valuesLength - 1].isEmpty();
+                if (hasMeaninglessTrailingDelimiter) {
+                    // adjust value length to account for the trailing delimiter
+                    valuesLength -= 1;
```

Comment:
We compensate for this code https://github.com/cBioPortal/cbioportal-core/blob/rfc79/src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java#L107
that forms csv row with a trailing comma. Even if we fix that code, DBs out there would contain rows with the trailing comma still. So this code is written to work with rows with trailing commas and without.

BTW I've just realized that I don't know about the piece of code that would produce rows without trailing commas. So maybe we should simplify this logic and always expect CSV rows to have a trailing comma. Let me know what you think.

Reply:
I'm not sure what the right answer is. I think it may be good for the cBioPortal project as a whole to choose and document a standard for comma-separated data value formatting and that all such values be validated and rejected if they conflict with the standard. In general I think it is a worse option to try to accept / accommodate two different flavors of format. If I needed to set the standard I think my personal preference would be to interpret a trailing comma to always indicate that there was another null value at the end of the list (after the final comma). So perhaps some values would be considered errors under that standard. But this is only my opinion/style preference. So I think the topic can be followed up on later.